lib/Wasmtime.hs

```diff
@@ -1122,7 +1123,7 @@ readMemoryAt ctx mem offset len = do
       else do
         res <- unsafeWithMemory ctx mem $ \mem_data_ptr mem_size ->
           BI.create mem_size $ \dst_ptr ->
-            BI.memcpy (advancePtr dst_ptr (fromIntegral offset)) mem_data_ptr mem_size
+            BI.memcpy dst_ptr (advancePtr mem_data_ptr (fromIntegral offset)) mem_size
```

Comment:
I meant to place the above comment here. 

Would it make sense to have a separate (newtype?) for offset and length respectively?